src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php

```diff
@@ -1061,25 +1062,54 @@ public function testMessengerWithMultipleBuses()
 
         $this->assertTrue($container->has('messenger.bus.commands'));
         $this->assertSame([], $container->getDefinition('messenger.bus.commands')->getArgument(0));
-        $this->assertEquals([
-            ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.commands']],
-            ['id' => 'reject_redelivered_message_middleware'],
-            ['id' => 'dispatch_after_current_bus'],
-            ['id' => 'failed_message_processing_middleware'],
-            ['id' => 'send_message', 'arguments' => [true]],
-            ['id' => 'handle_message', 'arguments' => [false]],
-        ], $container->getParameter('messenger.bus.commands.middleware'));
+
+        if (class_exists(DeduplicateMiddleware::class)) {
+            $this->assertEquals([
+                ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.commands']],
+                ['id' => 'reject_redelivered_message_middleware'],
+                ['id' => 'dispatch_after_current_bus'],
+                ['id' => 'failed_message_processing_middleware'],
+                ['id' => 'deduplicate_middleware'],
+                ['id' => 'send_message', 'arguments' => [true]],
+                ['id' => 'handle_message', 'arguments' => [false]],
+            ], $container->getParameter('messenger.bus.commands.middleware'));
+        } else {
```

Comment:
update the test deps next time to avoid this exra conditions :)

Reply:
it's all maintenance

Reply:
There’s not really a need to forbid using FrameworkBundle with Messenger < 7.3. But that would be the consequence if we cannot test it.

Reply:
i dont like the if/else condition based on class presence generally

Reply:
7.2 is tested without class X
7.3 is tsted with class X

it's that simple to me.

Reply:
> we would require people to update FrameworkBundle and components at the same time which would make updating harder

generally yes, it that prevents if/else flows in your codebase. We should prefer branches then.